source-mysql/main.go

```diff
@@ -66,9 +71,11 @@ type Config struct {
 	User     string `json:"user" jsonschema:"default=flow_capture,description=Database user to connect as."`
 	Password string `json:"password" jsonschema:"description=Password for the specified database user."`
 	DBName   string `json:"dbname" jsonschema:"description=Name of the database to connect to."`
-	ServerID int    `json:"server_id" jsonschema:"description=Server ID for replication."`
+	ServerID int    `json:"server_id" jsonschema:"description=Server ID for replication (each node in a cluster should have a unique ID)."`
 
 	WatermarksTable string `json:"watermarks_table,omitempty" jsonschema:"default=flow.watermarks,description=The name of the table used for watermark writes during backfills. Must be fully-qualified in '<schema>.<table>' form."`
+
+	SkipBinlogRetentionCheck bool `json:"skip_binlog_retention_check,omitempty" jsonschema:"default=false,description=For advanced users only. Bypasses the 'dangerously short binlog retention' sanity check at startup."`
```

Comment:
I think this was a good idea to add something to the config to allow disabling the check. Just to toss out an idea, this might be better as `ensure_minimum_binlog_retention`, which accepts a duration and always errors if the actual retention is less than the given value. Then we could use a default value of 7 days.

Reply:
I considered making it configurable, but I'm worried that having that value be configurable will lead users to want to tweak it.

Specifically I'm worried about the situation where a not-especially-knowledgeable user is setting up a capture from a preexisting database (especially one that they may not have authority to tweak, or not really feel comfortable configuring). If they see this check failing, there's going to be a strong temptation to just lower the setting to whatever value makes it pass, with shaky reasoning like "well the database has been running fine with that setting until now". Which, sadly, is setting themselves up for more difficult problems down the line.

I could see some utility in setting the threshold _higher_ than 7 days, but the parameter description would get pretty complicated at that point (something like "Specifies the minimum binlog retention period which will be accepted by the connector, with 0 meaning to skip the check entirely, and nonzero values <7 days adjusted up to 7 days for your own good"), and I think users that want to explicitly make it larger are already thinking about the right things and will probably get it right on their own.

Reply:
Makes sense :+1: